src/query/interpreter.cpp

```diff
@@ -1971,12 +1970,25 @@ constexpr auto ToStorageIsolationLevel(const IsolationLevelQuery::IsolationLevel
   }
 }
 
-PreparedQuery PrepareIsolationLevelQuery(ParsedQuery parsed_query, bool in_explicit_transaction,
+constexpr auto ToStorageMode(const StorageModeQuery::StorageMode storage_mode) noexcept {
+  switch (storage_mode) {
+    case StorageModeQuery::StorageMode::IN_MEMORY_TRANSACTIONAL:
+      return storage::StorageMode::IN_MEMORY_TRANSACTIONAL;
+    case StorageModeQuery::StorageMode::IN_MEMORY_ANALYTICAL:
+      return storage::StorageMode::IN_MEMORY_ANALYTICAL;
+  }
+}
+
+PreparedQuery PrepareIsolationLevelQuery(ParsedQuery parsed_query, const bool in_explicit_transaction,
                                          InterpreterContext *interpreter_context, Interpreter *interpreter) {
   if (in_explicit_transaction) {
     throw IsolationLevelModificationInMulticommandTxException();
   }
 
+  if (interpreter_context->db->GetStorageMode() == storage::StorageMode::IN_MEMORY_ANALYTICAL) {
+    throw IsolationLevelModificationInAnalyticsException();
+  }
+
   auto *isolation_level_query = utils::Downcast<IsolationLevelQuery>(parsed_query.query);
```

Comment:
what if after this if user concurrently modifies storage mode to analytical?

Reply:
Nothing will break, but user will succeed in attempt, and when changes back to in memory transactional, it will be in that state. This is something that @as51340 suggested. I am not sure if we should sync this part, but I can move this check inside changing isolation level, and then we are sure.

Reply:
Good catch.

Reply:
I think it's good as it is currently since Storage object isn't used in this method. Which check would you move where @antoniofilipovic, I don't get it?

Reply:
@as51340 I might be wrong but I think it is not good as it is, because I am still using storage_mode_ (GetStorageMode). This check should be moved to `SetIsolationLevel`